Chat app - Upgraded Version

tarelmadar opened the pull request:
I have upgraded the chat app to show messages history and some details about the user, such as username and timestamp.

tarelmadar committed:
Chat-App with all fixes

tarelmadar committed:
Upgraded Chat-App

ItzhakiGuy reviewed (dismissed):
Sending a new comment, I can't find the way to delete a comment :-)

ItzhakiGuy commented on sellotape/sellotape_dj/chat/models.py:
```diff
@@ -0,0 +1,16 @@
+
+# Create your models here.
+class Message(models.Model):
+    author = models.ForeignKey(User, related_name = 'author_messages', on_delete = models.CASCADE)
```
We have extended the regular Django `User` attributes, and we have called it a `Profile`, it should correspond with the `author` attribute on your `models.py` - chat app.
Plus, I think the `Message` object should be on the README.MD as you are creating a new ORM.
See PR #27 , it supposed to clarify the objects modeling.

Omeramsc replied:
Do note that if you change this, you'd also need a new migration.

11galpid replied:
@tarelmadar Did you address this?

Omeramsc commented on sellotape/sellotape_dj/chat/templates/chat/index.html:
```diff
@@ -0,0 +1,24 @@
+<html>
```
Is there a specific reason not to make use of the base.html template you guys implemented?

Omeramsc commented on sellotape/sellotape_dj/chat/templates/chat/room.html:
```diff
@@ -0,0 +1,94 @@
+{% load static %}
+<html>
```
Is there a specific reason not to make use of the base.html template you guys implemented?

Omeramsc commented on sellotape/sellotape_dj/chat/views.py:
```diff
@@ -0,0 +1,14 @@
+from django.shortcuts import render
+import json
+
+def index(request):
```
You better lint/reformat your code, 
Once you'll implement a CI with a linter, it'll fail in cases like this where there are no 2 lines between functions

11galpid replied:
We'll add a reformat PR to address this format issues, it happens in other parts of the app aswell.

Omeramsc commented on sellotape/sellotape_dj/chat/views.py:
```diff
@@ -0,0 +1,14 @@
+
+def index(request):
+    return render(request, 'chat/index.html')
+@login_required
```
^same

Omeramsc commented on sellotape/sellotape_dj/chat/tests.py:
```diff
@@ -0,0 +1,3 @@
+from django.test import TestCase
+
+# Create your tests here.
```
This is what we call "הכנה למזגן"... if you don't plan to introduce tests on this PR (although you probably should), this file is not needed.

11galpid committed:
General fixes to existing features
- Fixed messages were in the wrong order
- Added missing 'room_name' field to Message model so we can identify
  where each message came from
- Added the mapping to the relevant parts to use the 'room_name'
- Created migration files for this fix, old messages will be mapped to
  'old_room' room (because field didn't exist before)
- Generalized 'last_10_messages()' to 'last_messages()', number of
  messages can be more dynamic, and also choose the proper room

11galpid commented:
Pushed in a few small fixes, doesn't change the general structure of the chat-app

11galpid commented:
@tarelmadar Note, the 'Bootstrap the ChatApp' issue will be addressed in a different PR, after the chatapp will be approved formally, its not part of this PR

ItzhakiGuy reviewed (approved):
LGTM :-)

11galpid reviewed (approved):
Looks good
